packages/EasyApiToken/src/External/JwtTrait.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyApiToken\External;
+
+use Firebase\JWT\JWT as BaseJWT;
+use ReflectionMethod;
+
+/**
+ * @deprecated compatibility layer, will be removed in 5.0
+ * @internal
+ */
+trait JwtTrait
+{
+    private static bool $isFirebaseJwtV5;
+
+    /**
+     * @deprecated compatibility layer, will be removed in 5.0
+     * @internal
+     */
+    private static function isFirebaseJwtV5(): bool
+    {
+        if (isset(self::$isFirebaseJwtV5) === false) {
+            $methodParams = \array_map(
+                static fn ($parameter) => $parameter->name,
+                (new ReflectionMethod(BaseJWT::class, 'decode'))->getParameters()
```

Comment:
Hm, maybe it would be better to use smth like `Composer\InstalledVersions::getVersion($packageName)`?
I don't like to have reflection usage on each request.
wdyt?

Reply:
Then we will need extra packages.
@natepage do you mind?

<img width="30%" alt="Снимок экрана 2023-02-20 в 16 21 16" src="https://user-images.githubusercontent.com/6824784/220147727-b1397740-c7bf-40e2-814f-9a65cffac6c3.png">

<img width="30%" alt="Снимок экрана 2023-02-20 в 16 21 03" src="https://user-images.githubusercontent.com/6824784/220147716-d4901313-5a5e-4c4c-b7aa-696523dfdf76.png">

Reply:
We don't need these packages. You can use `version_compare` PHP function to compare version you got from `Composer\InstalledVersions::getVersion($packageName)`.

Reply:
At least we need `composer-runtime-api` for `Composer\InstalledVersions::getVersion($packageName)`

Reply:
But, looking to its description: 
```
composer-runtime-api 2.0 introduced a new Composer\InstalledVersions class
```
and looking to ours `vendor/composer` folder - we already have it.

Reply:
Usually in such situation I try to identify an easy thing that differentiate 2 versions of the same package, a new class, a missing class, method, etc. So it's quite easy to understand which version of the package you're dealing with by checking if the class exists for example.

Reply:
@ERuban, beyond belief, it works.